catalog/spatial/wps/spatial-wps-endpoint/src/main/java/org/codice/ddf/spatial/ogc/wps/process/endpoint/WpsEndpoint.java

```diff
@@ -0,0 +1,532 @@
+        } catch (RuntimeException e) {
+          LOGGER.debug("Error getting process offering from processRepository.", e);
+        }
+      }
+    }
+    Validator.validateDescribeRequest(processIds, processOfferings);
+    return processOfferings;
+  }
+
+  /**
+   * @param describeProcess
+   * @return
+   * @throws WpsException
+   */
+  @Override
+  public ProcessOfferings describeProcess(DescribeProcess describeProcess) {
+    Set<String> processIds =
+        describeProcess
+            .getIdentifier()
+            .stream()
+            .map(CodeType::getValue)
+            .collect(Collectors.toSet());
+    if (processIds.isEmpty()) {
+      processIds = ALL;
+    }
+    return describeProcess(processIds);
+  }
+
+  // todo - (RWY) add some sort of permissions check, may need to bubble it down into the algo's and
+  // add to describe
+
+  /**
+   * @param executeRequestType
+   * @return
+   * @throws WpsException
+   */
+  @Override
+  public Response execute(ExecuteRequestType executeRequestType) {
+    String processId = executeRequestType.getIdentifier().getValue();
+    Process process = getProcess(processId);
+
+    ExecutionDescription executionDescription = process.getExecutionDescription();
+    ExecutionRequest executionRequest =
+        wpsObjectConverter.getExecutionRequest(executionDescription, executeRequestType);
+    Validator.validate(executionDescription, executionRequest);
+    Operation operation = wpsObjectConverter.getOperation(process, executeRequestType.getMode());
+    if (operation == Operation.SYNC_EXEC) {
+      ProcessResult processResult;
+      processResult = process.syncExecute(executionRequest);
+      if (executionRequest.isRawResponse()) {
+        Literal literal =
+            processResult
+                .getOutput()
+                .stream()
+                .filter(Literal.class::isInstance)
+                .map(Literal.class::cast)
+                .findFirst()
+                .orElseThrow(
+                    () ->
+                        new WpsException(
+                            "", "InternalServerError", "", HttpURLConnection.HTTP_INTERNAL_ERROR));
+
+        return Response.ok(literal.getValue())
+            .encoding(literal.getFormat().getEncoding())
+            .type(literal.getFormat().getMimeType())
+            .build();
+      }
+      return Response.ok(wpsObjectConverter.createResult(processResult)).build();
+    } else {
+      ProcessStatus processStatus = process.asyncExecute(executionRequest);
+      return Response.ok(wpsObjectConverter.createStatusInfo(processStatus)).build();
+    }
+  }
+
+  /**
+   * @param jobId
+   * @return
+   * @throws WpsException
+   */
+  @Override
+  public StatusInfo getStatus(String jobId) {
+    ProcessStatus status =
+        processMonitors
+            .stream()
+            .map(
+                am -> {
+                  try {
+                    return am.getRequestStatus(jobId).get();
```

Comment:
❗️  Isn't this an optional? We probably should not be doing a blind get -- but instead return the optional (And don't return null from this method if we're returning optionals only) and then do a filter(Optional::isPresent) and map(Optional::get) (same goes for the other usages like this)

Reply:
For this, I don't see how I can get away from eventually having to do a get. However, I can move it to the end